TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/BlockLibrary/VirtualTexturePropertyBlockBuilder.cs

```diff
@@ -0,0 +1,153 @@
+using System.Collections.Generic;
+using UnityEditor.ShaderFoundry;
+using static UnityEditor.ShaderFoundry.VirtualTextureLayerAttribute;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    internal class VirtualTexturePropertyBlockBuilder : BasePropertyBlockBuilder
+    {
+        public class LayerData
+        {
+            internal string UniformName;
+            internal string DisplayName;
+            internal string TextureName;
+            internal LayerTextureType? TextureType = null;
+        }
+
+        const string VirtualTexturePropertyTypeName = "VTPropertyWithTextureType";
+        public int LayerCount = 2;
+        public int LayerToSample = 0;
+        
+        public List<LayerData> Layers = new List<LayerData>
+        {
+            new LayerData(),
+            new LayerData(),
+            new LayerData(),
+            new LayerData(),
+        };
+
+        public static string GetName(int layerIndex, string layerValue, string propertyValue, string fieldName)
```

Comment:
Maybe it would be better to pre-create all of them right off instead of potentially constructing each time?

Reply:
So this is for the built data. In particular, a case like:
```
[Properyt(uniformName = "MyUniform"][VirtualTexture(layerCount = 3)]
[VirtualTextureLayer(layerIndex = 0, displayName = "MyUniform0"]
VTProperty MyVT;
```
All layer uniforms will be changed to `MyUniform{i}` due to the property attribute.
Layer 0's display name is overridden to "MyUniform0", but all of the other layers will fallback to the default scheme which is based on the field name: `MyVT_Layer{i}`.

This is just encapsulating this data for unit test queries. The current layer data is used to build the property, but I don't cache the built results. I could do that as a separate field if you think it'd be cleaner .

Reply:
I will be switching this to a cache due to some other changes.